main.cpp

```diff
@@ -184,6 +190,12 @@ static void hue_to_rgb(double color_index, double num_colors, uint8_t& r, uint8_
     }
 }
 
+static void signalHandler(int)
+{
+	cleanup();
```

Comment:
I don't think `cleanup` is safe for a signal handler: https://man7.org/linux/man-pages/man7/signal-safety.7.html
Would't it be better to set a boolean exit flag and quit the main loop in that case?

Reply:
Nice idea. Absolutely.

Reply:
@lhecker i've now been rewriting `write_console`. So for UNIX that is safe. But for Windows I don't know. Do you still prefer a flag and having the cleanup call moved out into main?